shell/platform/linux/fl_window_state_monitor.h

```diff
@@ -0,0 +1,41 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_LINUX_FL_WINDOW_STATE_MONITOR_H_
+#define FLUTTER_SHELL_PLATFORM_LINUX_FL_WINDOW_STATE_MONITOR_H_
+
+#include <gtk/gtk.h>
+
+#include "flutter/shell/platform/linux/public/flutter_linux/fl_engine.h"
+
+G_BEGIN_DECLS
+
+G_DECLARE_FINAL_TYPE(FlWindowStateMonitor,
+                     fl_window_state_monitor,
+                     FL,
+                     WINDOW_STATE_MONITOR,
+                     GObject);
+
+/**
+ * FlWindowStateMonitor:
+ *
+ * Monitors a GtkWindow and reports state change events to the Flutter engine.
+ */
```

Comment:
Should this comment go above the `G_DECLARE_FINAL_TYPE` macro?

Reply:
We haven't in (at least most) other cases. Note GTK docs are pulled out by the first line (i.e. `FlWindowStateMonitor:`) so their location in the file doesn't matter. In other GObject code this is above the structure, which is not present in the header for final types.